src/analytics/ledger/unlock_conditions.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{CountValue, TransactionAnalytics};
+use crate::{
+    db::collections::analytics::UnlockConditionAnalyticsResult,
+    types::{
+        ledger::{LedgerOutput, LedgerSpent, MilestoneIndexTimestamp},
+        stardust::block::Output,
+    },
+};
+
+#[derive(Clone, Debug, Default)]
+#[allow(missing_docs)]
+pub struct UnlockConditionAnalytics {
+    pub timelock: CountValue,
+    pub expiration: CountValue,
+    pub storage_deposit_return: CountValue,
+    pub storage_deposit_return_inner_value: u64,
+}
+
+impl UnlockConditionAnalytics {
+    /// Initialize the analytics be reading the current ledger state.
+    pub fn init<'a>(unspent_outputs: impl IntoIterator<Item = &'a LedgerOutput>) -> Self {
+        let mut measurement = Self::default();
+        for output in unspent_outputs {
```

Comment:
Since this is a lot of code, and we do the same thing below, maybe we should put it in a separate fn?

Reply:
Actually, we could just use the init fn to build up an object then add/subtract it from self